### DISCUSSION TITLE
cmake xcb

### DISCUSSION THREAD

**Discussion by blobfish** [vsg-dev/VulkanSceneGraph#997] [Q&A]
I have built and installed vsg and everything appears to be working good. However from an external project, I can't link to the vsg dependency xcb. In my '/usr/local/lib/cmake/vsg/vsgTargets.cmake' I see the following lines:
```
set_target_properties(vsg::vsg PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "Vulkan::Vulkan;Threads::Threads;\$<LINK_ONLY:PkgConfig::xcb>"
)
```
"\$<LINK_ONLY:PkgConfig::xcb>" doesn't look right to me. Isn't that something that is suppose to be resolved at build time?

**Comment by robertosfield** [comment]
That looks the same as my vsgTargets.cmake haven't seen any problems,  This is mine:

~~~ cmake
set_target_properties(vsg::vsg PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "Vulkan::Vulkan;Threads::Threads;\$<LINK_ONLY:PkgConfig::xcb>"
)
~~~

CMake generates the vsgTargets.cmake for us.  I presume it's using this because we are using pkgconfig to get the details for xcb during the VSG build.

Do vsgXchange and vsgExamples work OK?

What OS/build tools are you using?

FYI, I have Ubuntu 22.04 and Kubuntu 22.04 on my laptop and desktop systems.

**Comment by blobfish** [comment]
thanks, good to know my file is the same and I didn't mess up the build somehow.

Yes, everything from VSG works as expected.

I am on debian bookworm using meson build. I will look into meson's cmake usage deeper. Thanks.

**Comment by blobfish** [comment]
Yeah this looks like a feature of cmake that meson currently isn't supporting. I will close this.
